inheritance of settings

omarwllorente opened the pull request:
Main changes are related to the inheritance of the options defined in settings.py though the rest of the code. Other minor changes to clarify name variables (case of key_salt in salted_hmac).

Most of the tests pass, but 3 (1 failure: test_encrypt_decrypt; and 2 errors: test_decryptor_invalid_token and test_unpadder_invalid_token, that both raise BadSignature errors instead of InvalidToken error).

Thanks.

georgemarshall commented on django_cryptography/utils/crypto.py:
```diff
@@ -18,30 +18,28 @@ class InvalidToken(Exception):
     pass
 
 
-def salted_hmac(key_salt, value, secret=None):
```
This refactor breaks parity with [django.utils.crypto.salted_hmac](https://github.com/django/django/blob/master/django/utils/crypto.py#L29), the goal is to have a "drop in" replacement.

omarwllorente replied:
Then shouldn't we define a CRYPTOGRAPHY_SIGNING_KEY in order to keep parity? This way, we would be able to maintain or change the default behaviour without ignoring what is defined in CRYPTOGRAPHY_KEY. It is my humble opinion, what do you think?

georgemarshall commented on django_cryptography/core/signing.py:
```diff
@@ -90,7 +91,7 @@ def loads(s, key=None, salt='django.core.signing', serializer=JSONSerializer, ma
 class Signer(object):
```
This also seems to deviate from parity with [django.core.signing.Signer](https://github.com/django/django/blob/master/django/core/signing.py#L152)

omarwllorente commented:
Hi George, I've changed the code to keep parity with django signing. I've added a signing variable called CRYPTOGRAPHY_SIGINGKEY which defaults to settings.SECRET_KEY if not specified in settings. Feel free to coment what you feel about it. Thank you for your time.
